pkgs/development/python-modules/zope-index/default.nix

```
@@ -0,0 +1,66 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  setuptools,
+  wheel,
+
+  # dependencies
+  btrees,
+  persistent,
+  zope-interface,
+
+  # tests
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "zope-index";
+  version = "8.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "zopefoundation";
+    repo = "zope.index";
+    tag = "${version}";
+    hash = "sha256-dHapd/+pJh6qzVx9FGSMmPsGbz8NhAoPqufXm3FOuM8=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "setuptools == 78.1.1" "setuptools"
```

```suggestion
      --replace-fail "setuptools ==" "setuptools >="
```
makes future updates easier